#DSC 320 Week 6

Chana Neuhaus
7/15/2026
DSC 320 Bellevue University

In [1]:
#importing dependencies
import pandas as pd
import numpy as np
from sklearn.preprocessing import StandardScaler, MinMaxScaler

##1. Data Normalization

In [2]:
#creating normalizer function
def normalize(vector):
  normalized_vector = (vector - np.min(vector))/(np.max(vector)-np.min(vector))
  return normalized_vector

##2. Data Standardization

In [3]:
#creating standardization function
def standardize(vector):
  mean = np.average(vector)
  stdev = np.std(vector)
  standardized_vector = (vector - mean)/stdev
  return standardized_vector

##3. Working with a dataframe

In [4]:
#importing dataset as pandas dataframe
df = pd.read_csv("calif_housing_data.csv")

####a) How many rows does this dataset have?

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   housing_median_age  20640 non-null  int64  
 1   total_bedrooms      20433 non-null  float64
 2   households          20640 non-null  int64  
 3   median_income       20640 non-null  float64
 4   median_house_value  20640 non-null  float64
dtypes: float64(3), int64(2)
memory usage: 806.4 KB


As we see can see from Range Index, the dataset has 20640 rows (the first of which contains the column names)

####b) What is the target vector for your model?

In [6]:
df.head()

,housing_median_age,total_bedrooms,households,median_income,median_house_value
0,41,129.0,126,8.3252,452600.0
1,21,1106.0,1138,8.3014,358500.0
2,52,190.0,177,7.2574,352100.0
3,52,235.0,219,5.6431,341300.0
4,52,280.0,259,3.8462,342200.0


We can assume the target variable is median_house_value. With this, we could create a regression model that predicts the value for a house in California given these features.

####c) Create a new feature by taking the total bedrooms divided by the number of households. What
does this new feature represent?

In [7]:
df["mean_bedrooms"] = df["total_bedrooms"] / df["households"]
df.head()

,housing_median_age,total_bedrooms,households,median_income,median_house_value,mean_bedrooms
0,41,129.0,126,8.3252,452600.0,1.023810
1,21,1106.0,1138,8.3014,358500.0,0.971880
2,52,190.0,177,7.2574,352100.0,1.073446
3,52,235.0,219,5.6431,341300.0,1.073059
4,52,280.0,259,3.8462,342200.0,1.081081


The new column represents the average number of bedrooms per house on the block.

####d)

In [8]:
#creating new DataFrame
new_df = pd.DataFrame([df["housing_median_age"],df["median_income"], df["mean_bedrooms"]]).transpose()
print(new_df.info())
new_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20640 entries, 0 to 20639
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   housing_median_age  20640 non-null  float64
 1   median_income       20640 non-null  float64
 2   mean_bedrooms       20433 non-null  float64
dtypes: float64(3)
memory usage: 483.9 KB
None


,housing_median_age,median_income,mean_bedrooms
0,41.0,8.3252,1.023810
1,21.0,8.3014,0.971880
2,52.0,7.2574,1.073446
3,52.0,5.6431,1.073059
4,52.0,3.8462,1.081081


e)

In [9]:
#applying standardization function to every column, and adding new standardized column to dataset
#using .ffill to fill NaN values
for column in new_df.columns:
  new_df["standardized_" + column] = standardize(new_df[column].ffill())
new_df.head()

,housing_median_age,median_income,mean_bedrooms,standardized_housing_median_age,standardized_median_income,standardized_mean_bedrooms
0,41.0,8.3252,1.023810,0.982143,2.344766,-0.153946
1,21.0,8.3014,0.971880,-0.607019,2.332238,-0.263491
2,52.0,7.2574,1.073446,1.856182,1.782699,-0.049236
3,52.0,5.6431,1.073059,1.856182,0.932968,-0.050053
4,52.0,3.8462,1.081081,1.856182,-0.012881,-0.033131
